test/service_test.cc

```diff
@@ -111,6 +113,44 @@ class ServiceTest : public TestWithParam<Envoy::Network::Address::IpVersion> {
   int grpc_server_port_{0};
 };
 
+class ServiceTestWithParameterizedConstructor : public ServiceTest {
+public:
+  void SetUp() override {
+    auto logging_context = std::make_unique<Envoy::Logger::Context>(
+        spdlog::level::info, "%L %n [%g:%#] %v", log_lock_, false);
+    service_ = std::make_unique<ServiceImpl>(std::move(logging_context));
+    grpc::ServerBuilder builder;
+    loopback_address_ = Envoy::Network::Test::getLoopbackAddressUrlString(GetParam());
+
+    builder.AddListeningPort(fmt::format("{}:0", loopback_address_),
+                             grpc::InsecureServerCredentials(), &grpc_server_port_);
+    builder.RegisterService(service_.get());
+    server_ = builder.BuildAndStart();
+    setupGrpcClient();
+    setBasicRequestOptions();
+  }
+
+  Envoy::Thread::MutexBasicLockable log_lock_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ServiceTestWithParameterizedConstructor,
+                         ValuesIn(Envoy::TestEnvironment::getIpVersionsForTest()),
+                         Envoy::TestUtility::ipTestParamsToString);
+
+// Test single NH run with parameterized constructor
+TEST_P(ServiceTestWithParameterizedConstructor, BasicWithUserSpecifiedLoggingContext) {
+  auto r = stub_->ExecutionStream(&context_);
```

Comment:
Two things here:
1. We try to avoid using auto for readability reasons, unless it is incredibly obvious what the returning type is (such as when the constructor already indicates the type or we're iterating through an array where the type is immediately present there).  See the "Decision" at https://google.github.io/styleguide/cppguide.html#Type_deduction

2. We also try to avoid abbreviations in general.  If this is a request, then we'd usually call the variable request instead of r.  See https://google.github.io/styleguide/cppguide.html#General_Naming_Rules